pkg/apis/build/v1alpha1/trigger_when.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package v1alpha1
+
+// TriggerWhen a given scenario where the webhook trigger is applicable.
+type TriggerWhen struct {
+	// Name name or the short description of the trigger condition.
+	Name string `json:"name"`
+
+	// Type the event type, the name of the webhook event.
+	Type WhenTypeName `json:"type,omitempty"`
```

Comment:
If this is required, should we remove omitempty ?

Reply:
Thanks, indeed the `omitempty` must not be there.